### ISSUE TITLE
Expose HVACMode.DRY for climate devices in HomeKit

### ISSUE THREAD

**Issue by Stetocasa** [oyvindwe/connectlife-ha#132]
Hello, first of all thanks for your precious job and the fantastic integration.

I'm writing to understand how expose to HomeKit (trough the official integration) the possibility to control the dehumidifier.

In home assistant, this is possible trough the main panel as below
<img width="495" alt="Screenshot 2024-09-02 alle 10 43 07" src="https://github.com/user-attachments/assets/5b6fad53-d812-4157-bc72-45bcb6775fc2">
 but when exposed to HomeKit is just settable, cool, hot, and automatic (+ an additional switch to control the fan speed)

<img width="340" alt="Screenshot 2024-09-02 alle 10 44 55" src="https://github.com/user-attachments/assets/a8d578e1-f26a-418d-af8b-0c6f12d48696">

"Non attivo" = OFF
" Freddo" = Cool
" Caldo" = Hot
" Automatico" = Automatic 

and control fan trough the dimmer sensor with the blue icon

Basically just dehumidifier are missing

Any chance in your integration to have a separate switch (or control) for this mode?

Thanks for your feedback

**Comment by oyvindwe** [comment]
This will require some digging in the HomeKit bridge integration in HA.

Which device do you have?

**Comment by Stetocasa** [comment]
> This will require some digging in the HomeKit bridge integration in HA.
> 
> Which device do you have?

Hisense air conditioner

**Comment by oyvindwe** [comment]
In any case, this looks like a limitation in HomeKit - both modes `HVACMode.COOL` and `HVACMode.DRY` in Home Assistant are mapped to `HC_HEAT_COOL_COOL` in HomeKit: https://github.com/home-assistant/core/blob/dev/homeassistant/components/homekit/type_thermostats.py#L135-L143

I'm not familiar with the HomeKit API, but it may look like Apple needs to add support for more modes.

A possible workaround is to install [humidifier_template](https://github.com/tsunglung/hass_humidifier_template), and expose your air conditioner as a dehumidifier device as well ro HomeKit. (HomeKit supports dehumidifiers - I can access my Tuya dehumidifier in HomeKit via Home Assistant.)

**Comment by Stetocasa** [comment]
![IMG_3225](https://github.com/user-attachments/assets/7d7b02ac-58a6-43da-818d-3d4c88142632)
This is the hardware
The request is to allow any such of control in HomeKit to control the Dry mode.
Maybe I was inaccurate to explain what I’m looking for.
As you said, on thermostat HomeKit allow just cool and hot, the dry mode should be launched just creating a normal switch control 
In your integration is the icon marked here 
![IMG_3227](https://github.com/user-attachments/assets/a98c470c-c554-4cd2-9dd0-480339ec6d18)

**Comment by oyvindwe** [comment]
> As you said, on thermostat HomeKit allow just cool and hot, the dry mode should be launched just creating a normal switch control

I see. Unfortunately it's not as simple as making a separate switch in the integration, as the modes in Home Assistant are mapped different values of `t_work_mode`:
```
  - property: t_work_mode
    climate:
      target: hvac_mode
      options:
        0: fan_only
        1: heat
        2: cool
        3: dry
        4: auto
```

I'm not sure how the logic should be for a `dry` switch. Switching it on would have to set `t_work_mode` to 3, but what should switching it off do - set to 2 (`cool`) perhaps?

You could try to create a [template switch](https://www.home-assistant.io/integrations/switch.template/) and expose that to Home Assistant - something like this perhaps (completely untested - I'm not very proficient with templating in Home Assistant):

```yaml
switch:
  - platform: template
    switches:
      ac_dry:
        value_template: "{{ is_state_attr('climate.mansarda', 'hvac_mode', 'dry') }}"
        turn_on:
          action: climate.set_hvac_mode
          data:
            hvac_mode: 'dry'
          target:
            entity_id: climate.mansarda
        turn_off:
          action: climate.set_hvac_mode
          data:
            hvac_mode: 'cool'
          target:
            entity_id: climate.mansarda
```

**Comment by Stetocasa** [comment]
This turnaround works, and now I can control Dry from HomeKit! Thanks !
![IMG_3230](https://github.com/user-attachments/assets/00142e89-0c07-4ad9-95f9-e1e63ecdf8d9)

nevertheless, if you can integrate this template in your integration, the dry switch can be grouped inside the air condition group as today you already did merging the thermostat with the “fan”‘to control the device fan speed 
![IMG_3231](https://github.com/user-attachments/assets/080546cf-a9e4-4ed7-96a1-dc3b78dfe511)

**Comment by oyvindwe** [comment]
Happy to hear that the switch template works!

Including a button for `dry` mode in the Climate entity card is a core feature of Home Assistant, not something this integration has added.

I don't think it's a good solution to add an extra switch for the `dry` mode in this integration just because HomeKit has this limitation. This is rather something that should be considered to be a part of the [HomeKit Bridge](https://www.home-assistant.io/integrations/homekit/) integration, as it affects all climate entities that supports both `cool` an `dry` modes.
